### ISSUE TITLE
Make autogenerated finders overridable (Skip def using Module.defines? before compile)

### ISSUE THREAD

**Issue by chrismccord** [chrismccord/atlas#6]

**Comment by scrogson** [comment]
Is this regarding the `with_field_name` finders? https://github.com/chrismccord/atlas/blob/master/lib/atlas/finders.ex#L30

What if we removed those in favor of a `find_by/1` which takes a keyword list like ActiveRecord 4?

``` elixir
User.find_by(username: "bob", password: "secret")
```

**Comment by chrismccord** [comment]
Yes, :+1: on the removal. We want to make sure only valid keys are passed.

**Comment by chrismccord** [comment]
ie, we should raise something like Schema.UndefinedFieldError if a passed key does not map to the defined schema

**Comment by scrogson** [comment]
gotcha.
